src/ds/logger.h

```diff
@@ -152,18 +212,30 @@ namespace logger
       ss << f;
       return *this;
     }
+
+    void finalize()
+    {
+      msg = ss.str();
+    }
   };
 
 #ifdef INSIDE_ENCLAVE
   struct Out
   {
     bool operator==(LogLine& line)
     {
+      line.finalize();
       if (line.log_level == Level::FATAL)
-        throw std::logic_error("Fatal: " + line.ss.str());
+        throw std::logic_error("Fatal: " + line.msg);
```

Comment:
This has removed context from fatal log messages, which are the ones we most want `file:line_no` for. Maybe we need something like a `default_format(time, level, file, line, message)`? We can call that here and also from the default std::cout outputter in `Out`.

Reply:
Alternative: `Logger` separates into 2 methods `write` and`format`, the default formatter becomes a `Logger` instance as well, and is pushed to `custom_loggers` first. `write` just calls each `custom_logger` in turn, and this first `write`s then throws `logic_error(loggers().first().format(…))`. If we start to rely on the JSON logs (or something more advanced, pushing logs directly to a remote service) then we'll eventually have situations where the std::cout log is ignored, unretrievable, and in that case we definitely want `FATAL` messages to be logged to whatever `Logger` we're actually using.

Reply:
We definitely don't want to lose information on the fatal lines!